kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -252,7 +261,8 @@ private long commitWithRetry(
                 protocol,
                 Optional.empty() /* winningTxnRowIdHighWatermark */,
                 dataActions,
-                resolvedDomainMetadatas);
+                resolvedDomainMetadatas,
```

Comment:
Can we do the check outside `updateRowIdHighWatermarkIfNeeded` so we don't even need to change the signature of `pdateRowIdHighWatermarkIfNeeded`